src/clojure_commons/file_utils.clj

```diff
@@ -130,12 +130,12 @@
         (if (.mkdir f) f (recur (inc idx))))
       nil)))
 
-(defn ^File temp-dir
+(defn temp-dir
   "Creates a temporary directory.  This function is used by the with-temp-dir
    macro to create the temporary directory."
-  ([prefix err-fn]
+  (^File [prefix err-fn]
      (temp-dir prefix (file (System/getProperty "user.dir")) err-fn))
-  ([prefix ^File parent err-fn]
+  (^File [prefix parent err-fn]
```

Comment:
I don't think that removing the type hint here will cause a problem, but I could be wrong.